### ISSUE TITLE
Possibly erroneous warning about `?` functions

### ISSUE THREAD

**Issue by rschuetzler** [exercism/elixir-analyzer#443]
Submitted this (probably bad) solution to the Nth Prime problem, and got the automated feedback:

"Elixir has specific naming conventions which include how to name [functions that return a boolean](https://hexdocs.pm/elixir/naming-conventions.html#trailing-question-mark-foo):

Functions that return a boolean are named with a trailing question mark."

Maybe I missed something, but I think my `is_prime?` function is the only one that returns a boolean, and it does, in fact, have a question mark.

```elixir
defmodule Prime do
  @doc """
  Generates the nth prime.
  """
  @spec nth(non_neg_integer) :: non_neg_integer
  def nth(1), do: 2

  def nth(count) when is_integer(count) and count > 0 do
    generate_primes(count)
  end

  defp generate_primes(n, primes \\ [2])
  defp generate_primes(n, primes) when length(primes) == n, do: List.first(primes)

  defp generate_primes(n, primes) do
    next_prime = find_next_prime(List.first(primes) + 1, primes)
    generate_primes(n, [next_prime | primes])
  end

  defp find_next_prime(number, primes) do
    if is_prime?(number, primes) do
      number
    else
      find_next_prime(number + 1, primes)
    end
  end

  defp is_prime?(2, [2]), do: true

  defp is_prime?(number, primes) when number > 2 do
    !Enum.any?(primes, &(rem(number, &1) == 0))
  end
end
```

**Comment by jiegillet** [comment]
I believe the issue is the very next item on the link you shared, [the prefix `is_` is reserved for guards](https://hexdocs.pm/elixir/naming-conventions.html#is_-prefix-is_foo). 
wasn't this shown at the end of the feedback?
 > Note that guards, some built-in functions and macros that return booleans and are valid in guards follow anoth    er convention and are named with a `is_` prefix (`def is_integer`, `defguard is_foo`, `defmacro is_foo`). Both     conventions must not mix.

**Comment by rschuetzler** [comment]
🤦 Ah, yes. It was there. I just stopped reading when I got to the code block because I had the question marks so I wasn't sure why it was telling me about them.
